hoodie-client/src/main/java/com/uber/hoodie/func/BufferedIteratorWrapper.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.uber.hoodie.func;
+
+import com.uber.hoodie.config.HoodieWriteConfig;
+import com.uber.hoodie.exception.HoodieException;
+import com.uber.hoodie.func.payload.BufferedIteratorPayloadFunction;
+import java.util.Iterator;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.function.Function;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.spark.TaskContext;
+import org.apache.spark.TaskContext$;
+
+/**
+ * Wrapper for a BufferedIterator operation. This class takes as input the input iterator which
+ * needs to be buffered, the runnable function that needs to be executed in the reader thread and
+ * return the transformed output based on the writer function
+ */
+public class BufferedIteratorWrapper<I, O, E> {
```

Comment:
lets rename to something more specific.. It feels like we are overloading `Wrapper` a lot. How about `BufferedIteratorExecutor`

Reply:
Add a test here?

Reply:
Addressed.